src/Microsoft.ML.Transforms/Text/TextCatalog.cs

```diff
@@ -62,7 +62,10 @@ public static class TextCatalog
             TextFeaturizingEstimator.Options options,
             params string[] inputColumnNames)
             => new TextFeaturizingEstimator(Contracts.CheckRef(catalog, nameof(catalog)).GetEnvironment(),
-                outputColumnName, inputColumnNames, options);
+                outputColumnName,
+                inputColumnNames == null || inputColumnNames.Length == 0 || (inputColumnNames.Length == 1 && inputColumnNames[0] == "")  ?
```

Comment:
>inputColumnNames[0] == "" [](start = 109, length = 25)

This should check for `!string.IsNullOrWhiteSpace` instead, but I am actually not sure if we should replace `inputColumnNames` with `new[] {outputColumnName}` in this case, might be better to throw (The constructor of `TextFeaturizingEstimator` will throw in this case, so no need to check for this here).

Reply:
@codemzs advised me to also cover the case when an empty string is passed as the only argument for inputColumnNames.

If it is not necessary to cover it, then I can simply remove the code and the test related to this case.

Reply:
I don't think this case is needed - we are talking about the case where a user explicitly passes an empty string with the intention of having the input column name be identical to the output column name? It doesn't seem necessary, and it doesn't say anywhere in the documentation that this can be done.

---
In reply to: [324773767](https://github.com/dotnet/machinelearning/pull/4211#discussion_r324773767) [](ancestors = 324773767)

Reply:
Ok. I will remove the code related to this case. Thanks for the feedback.